tool/plugin/lib/plugin.dart

```diff
@@ -1148,3 +1147,35 @@ class TestCommand extends ProductCommand {
     throw 'integration test execution not yet implemented';
   }
 }
+
+Future<String> makeDevLog(BuildSpec spec) async {
+  _checkGitDir();
+  var gitDir = await GitDir.fromExisting(rootPath);
+  var since = await lastRelease();
+  var processResult =
+      await gitDir.runCommand(['log', '$since..HEAD', '--oneline']);
+  String out = processResult.stdout;
+  var messages = out.trim().split('\n');
+  var devLog = StringBuffer();
+  devLog.writeln('## Changes since $since');
+  messages.forEach((m) {
+    devLog.writeln(m.replaceFirst(RegExp(r'^[0-9A-Fa-f]+ '), '- '));
+  });
+  devLog.writeln();
+  return devLog.toString();
+}
+
+Future<String> lastRelease() async {
+  _checkGitDir();
+  var gitDir = await GitDir.fromExisting(rootPath);
+  var processResult = await gitDir.runCommand(['branch', '--list', 'release*']);
```

Comment:
Is this output sorted in some way (chronologically)? It looks like below we do depend on a specific ordering.

Reply:
Also, will developers have these branches locally? These will be branches on the upstream repo.

Reply:
According to the git docs it is sorted by byte value. If we're still doing this in 5 years we will have a problem when we get to release_100.

I have the branches locally, as will anyone who does a build. But to run this as a service it will need to make sure the branch is checked out locally. If that's travis we'll have to modify the .travis.yml to include the release_* branch for a dev channel build -- if we use travis. Sounds like another change to `plugin generate` once we know where this will run.

Reply:
The issue as I see it is that people likely won't have all the remote branches available locally (this would happen if one person does M40 and another does M41, or, if someone hasn't driven a release for a while).

We may want to look at some additional git-fu here. If there's a remote repo named upstream, we should likely query the branches for the upstream fork. If there's no upstream, we can fall-back on querying the local branches.

Reply:
It looks like `-r` will show remote tracking branches. Those remote branches might not exist locally however if the users hasn't run `git fetch upstream`.

Reply:
I added a TODO here, and fixed plugin.xml generation. It was setting the untilVersion too low.
PTAL

Reply:
👍 lgtm